Code/Framework/AtomCore/AtomCore/Instance/InstanceDatabase.h

```diff
@@ -362,13 +362,19 @@ namespace AZ
         template<typename Type>
         Data::Instance<Type> InstanceDatabase<Type>::FindOrCreate(const Asset<AssetData>& asset, const AZStd::any* param)
         {
-            return FindOrCreate(Data::InstanceId::CreateFromAssetId(asset.GetId()), asset, param);
+            return FindOrCreate(Data::InstanceId::CreateFromAsset(asset), asset, param);
         }
 
         template<typename Type>
         Data::Instance<Type> InstanceDatabase<Type>::Create(const Asset<AssetData>& asset, const AZStd::any* param)
         {
-            return FindOrCreate(Data::InstanceId::CreateRandom(), asset, param);
+            // Handle the rare case where a random ID might already be in use
+            auto id = Data::InstanceId::CreateRandom();
+            while (Find(id))
+            {
+                id = Data::InstanceId::CreateRandom();
+            }
+            return FindOrCreate(id, asset, param);
```

Comment:
While it'd be incredibly unlikely to happen, you'd need a mutex here to be completely safe. Otherwise, it may be possible for a different thread to pick the same random id and pass the `Find(id)` check.